lib/concentrate/vehicle_position.ex

```diff
@@ -98,17 +108,76 @@ defmodule Concentrate.VehiclePosition do
               second.previous_vehicle_schedule_adherence_string,
               first.previous_vehicle_schedule_adherence_string
             ),
-          route_id: first_value(second.route_id, first.route_id),
+          route_id:
+            swiftly_priority(
+              VehiclePosition.source(second),
+              second.route_id,
+              VehiclePosition.source(first),
+              first.route_id
+            ),
           schedule_adherence_secs:
             first_value(second.schedule_adherence_secs, first.schedule_adherence_secs),
           schedule_adherence_string:
             first_value(second.schedule_adherence_string, first.schedule_adherence_string),
           scheduled_headway_secs:
-            first_value(second.scheduled_headway_secs, first.scheduled_headway_secs)
+            first_value(second.scheduled_headway_secs, first.scheduled_headway_secs),
+          source: merge_sources(first, second),
+          data_discrepancies: discrepancies(first, second)
       }
     end
 
     defp first_value(value, _) when not is_nil(value), do: value
     defp first_value(_, value), do: value
+
+    defp swiftly_priority("swiftly", nil, _, value), do: value
+
+    defp swiftly_priority("swiftly", value, _, _), do: value
+
+    defp swiftly_priority(other_source, other_value, "swiftly", value),
+      do: swiftly_priority("swiftly", value, other_source, other_value)
+
+    defp swiftly_priority(_, first_value, _, second_value),
+      do: first_value(first_value, second_value)
```

Comment:
If merging a `busloc|swiftly` source whose `trip_id` originally came from swiftly, with a more recent `busloc` assignment with a bad `trip_id`, this might take the `busloc` assignment over the `swiftly` one.